generators/generator-base-private.js

```diff
@@ -1463,6 +1463,34 @@ module.exports = class extends Generator {
         return primaryKeyType;
     }
 
+    /**
+     * Returns the primary key value based on authentication type, DB, given association and default value
+     *
+     * @param {string} authenticationType - the auth type
+     * @param {string} databaseType - the database type
+     * @param {any} relationships - relationships
+     * @param {string} defaultValue - default value
+     */
+    getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, defaultValue) {
+        const primaryKeyType = this.getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
+        let value;
+        switch (primaryKeyType) {
+            case 'String':
+                value = `"id${defaultValue}"`;
+                // Special case with a OneToOne relationship with User and @MapsId when using OAuth
+                if (databaseType === 'sql') {
+                    value = 'UUID.randomUUID().toString()';
+                }
+                break;
+            case 'UUID':
+                value = 'UUID.randomUUID()';
+                break;
+            default:
+                value = `${defaultValue}L`;
```

Comment:
Is the letter `L` at the end a typo?

Reply:
Don't think so, `L` may mean the value is a Long.